include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,556 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+#include <vk_mem_alloc.h>
+
+#include "utils/misc.h"
+#include "vkwrap/core.h"
+#include "vkwrap/error.h"
+
+#include <range/v3/algorithm/any_of.hpp>
+
+#include <range/v3/range/conversion.hpp>
+
+#include <array>
+#include <memory>
+#include <unordered_map>
+#include <utility>
+
+namespace std
+{
+
+template <> struct hash<vk::Buffer>
+{
+    size_t operator()( const vk::Buffer& buffer ) const
+    {
+        return reinterpret_cast<size_t>( buffer.operator VkBuffer() );
+    } // operator()
+
+}; // struct hash<vk::Buffer>
+
+template <> struct hash<vk::Image>
+{
+    size_t operator()( const vk::Image& image ) const
+    {
+        return reinterpret_cast<size_t>( image.operator VkImage() );
+    } // operator()
+
+}; // struct hash<vk::Image>
```

Comment:
Maybe this can be written as:
```cpp
return std::bit_cast<size_t>(buffer.get()); // Return pointer as hash
```

Reply:
About `std::bit_cast<size_t>` agree. Also there is no `get` method in `vk::Buffer`.

Reply:
I meant to say operator* instead of get.